### PR TITLE
Implementing `switch` expressions in `lib/src/material/`

### PR DESCRIPTION
**Pull request by nate-thegrate**
This PR is the 6ᵗʰ step in the journey to solve issue #136139 and make the entire Flutter repo more readable.

(previous pull requests: #139048, #139882, #141591, #142279, #142634)

The current focus is on `packages/flutter/lib/src/material/`. The previous 2 PRs covered files in this directory starting with letters `a-m`; this one takes care of everything else.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/radio_list_tile.dart [hunk 1]
```diff
@@ -485,15 +485,11 @@ class RadioListTile<T> extends StatelessWidget {
...
-      case ListTileControlAffinity.platform:
-        leading = control;
-        trailing = secondary;
-      case ListTileControlAffinity.trailing:
-        leading = secondary;
-        trailing = control;
-    }
+    (leading, trailing) = switch (controlAffinity) {
+      ListTileControlAffinity.leading ||
+      ListTileControlAffinity.platform => (control, secondary),
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/radio_list_tile.dart:490]: This looks strange, even though it is going to be a very long line, I would put this all in one.

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -2444,14 +2423,12 @@ class _TimePickerDialogState extends State<TimePickerDialog> with RestorationMix
...
-    switch (theme.materialTapTargetSize) {
-      case MaterialTapTargetSize.padded:
-        tapTargetSizeOffset = Offset.zero;
-      case MaterialTapTargetSize.shrinkWrap:
-        // _dialogSize returns "padded" sizes.
-        tapTargetSizeOffset = const Offset(0, -12);
-    }
+    final Offset tapTargetSizeOffset = switch (theme.materialTapTargetSize) {
+      MaterialTapTargetSize.padded => Offset.zero,
+
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/time_picker.dart:2428]: nit: remove this blank line?

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by Piinks — APPROVED** [review]
Also LGTM with the outstanding nits. :)

File: packages/flutter/lib/src/material/radio_list_tile.dart (hunk 1 above)
Comment (Piinks) [reply packages/flutter/lib/src/material/radio_list_tile.dart:490]:
```suggestion
      ListTileControlAffinity.leading || ListTileControlAffinity.platform => (control, secondary),
```

File: packages/flutter/lib/src/material/time_picker.dart (hunk 2 above)
Comment (Piinks) [reply packages/flutter/lib/src/material/time_picker.dart:2428]:
```suggestion
```

**Comment by Piinks** [comment]
> with the outstanding nits.

Which are now resolved. Thanks for contributing!

**Pull request merged**